issue: Email Templates Static

JediKev opened the pull request:
This addresses an issue reported on the Forum where cloning a Template Set gives you a fatal error of `Non-static method EmailTemplateGroup::add() cannot be called statically`. This updates the `add()` method to static for `EmailTemplateGroup` and `EmailTemplate` classes.

JediKev committed:
issue: Email Templates Static

This addresses an issue reported on the Forum where cloning a Template Set
gives you a fatal error of `Non-static method EmailTemplateGroup::add()
cannot be called statically`. This updates the `add()` method to static for
`EmailTemplateGroup` and `EmailTemplate` classes.

sdm216 commented:
This issue also exists for adding an email to banlist. 
`Non-static method Banlist::add() cannot be called statically`

JediKev committed:
issue: Banlist Non-Static

This addresses an issue where `Banlist::add()` and `Banlist::remove()` are
non-static methods that are being called statically. This updates the
methods to static so that they can be called statically.

JediKev commented:
@sdm216 Added, thanks.

JediKev committed:
issue: Auth_SASL Non-Static

This addresses an issue where `Auth_SASL::factory()` is being called
statically but is not static. This updates the method to static so it can be
called statically with no issues.
